wallet/src/actors/worker/methods.rs

```diff
@@ -307,6 +319,202 @@ impl Worker {
+                    Some(IndexTransactionQuery::InputTransactions(
+                        commit
+                            .body
+                            .collateral
+                            .iter()
+                            .map(|input| input.output_pointer().clone())
+                            .collect(),
+                    ))
+                }
+                types::Transaction::Tally(tally) => Some(IndexTransactionQuery::DataRequestReport(
+                    tally.dr_pointer.to_string(),
+                )),
+                _ => None,
+            })
+            .collect();
+
+        let query_results: Result<Vec<model::ExtendedTransaction>> = txns
+            .into_iter()
+            .zip(queries)
+            .map(|(transaction, opt_query)| match opt_query {
+                Some(query) => match query {
+                    IndexTransactionQuery::InputTransactions(pointers) => {
+                        Ok(model::ExtendedTransaction {
+                            transaction,
+                            metadata: Some(model::TransactionMetadata::InputValues(
+                                self.get_vt_outputs_from_pointers(&pointers)?,
+                            )),
+                        })
+                    }
+                    IndexTransactionQuery::DataRequestReport(dr_id) => {
+                        let retrieve_responses =
+                            async { self.query_data_request_report(dr_id).await };
+                        let report = futures03::executor::block_on(retrieve_responses)?;
+
+                        Ok(model::ExtendedTransaction {
+                            transaction,
+                            metadata: Some(model::TransactionMetadata::Tally(Box::new(report))),
+                        })
+                    }
+                },
+                None => Ok(model::ExtendedTransaction {
+                    transaction,
+                    metadata: None,
+                }),
+            })
+            .collect();
+
+        Ok(query_results?)
+    }
+
+    /// Retrieve Value Transfer Outputs of a list of Output Pointers (aka input fields in transactions).
+    pub fn get_vt_outputs_from_pointers(
+        &self,
+        output_pointers: &[types::OutputPointer],
+    ) -> Result<Vec<types::VttOutput>> {
+        // Query the node for the required Transactions
+        let txn_futures = output_pointers
+            .iter()
+            .map(|output| self.query_transaction(output.transaction_id.to_string()));
+        let retrieve_responses = async { futures03::future::try_join_all(txn_futures).await };
+        let transactions: Vec<types::Transaction> =
+            futures03::executor::block_on(retrieve_responses)?;
+        log::info!(
+            "Received {} wallet transactions from node",
+            transactions.len()
+        );
+
+        let result: Result<Vec<types::VttOutput>> = transactions
+            .iter()
```

Comment:
Same, you can probably remove all the `.map(types::VttOutput::clone)`

```suggestion
            .into_iter()
```

Reply:
Unfortunately, it seems not possible to get rid of the `clone`s :disappointed: